src/wx/widgets/sdljoy.cpp

```diff
@@ -314,24 +310,44 @@ void wxSDLJoy::Poll()
 
 void wxSDLJoy::ConnectController(uint8_t joy)
 {
+    SDL_Joystick* js_dev = nullptr;
+
     if (SDL_IsGameController(joy)) {
-        if (!(joystate[joy].dev = SDL_GameControllerOpen(joy))) {
+        auto dev = SDL_GameControllerOpen(joy);
+
+        if (!dev) {
             wxLogDebug("SDL_GameControllerOpen(%d) failed: %s", joy, SDL_GetError());
-            return;
+        }
+        else {
+            joystate[joy].dev = dev;
+
+            js_dev = SDL_GameControllerGetJoystick(dev);
         }
     }
     else {
-        if (!(joystate[joy].dev = SDL_JoystickOpen(joy))) {
+        if (!(js_dev = SDL_JoystickOpen(joy))) {
             wxLogDebug("SDL_JoystickOpen(%d) failed: %s", joy, SDL_GetError());
-            return;
         }
+        else {
+            joystate[joy].dev = js_dev;
+        }
+    }
+
+    if (js_dev) {
+        auto instance = SDL_JoystickInstanceID(js_dev);
+
+        instance_map[instance] = &(joystate[joy]);
+
+        joystate[joy].instance = instance;
     }
+
+    joystate[joy].index = joy;
```

Comment:
It looks like this will get executed even in a failure case.  Should this be moved to the slightly earlier if statement?

Reply:
This is intended, the state has to exist even if the device is not connected.